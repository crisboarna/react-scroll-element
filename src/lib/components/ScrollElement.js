// @flow
import React from 'react';
import PropTypes from 'prop-types';
import './ScrollElement.scss';

type Props = {
  type ?: string,
  text ?: string
};

const ScrollElement = ({type, text} : Props) => {
   return(
      <div className='scroll-element'>
        < div className={
          // $FlowFixMe: This type cannot be coerced to string
          `scroll-element-${type}`}>
          <span></span>{text}</div>
      </div>
    );
};

ScrollElement.propTypes = {
  type: PropTypes.string,
  text: PropTypes.string
};

ScrollElement.defaultProps = {
  type: 'chevron',
  text: 'Scroll'
};

export default ScrollElement;