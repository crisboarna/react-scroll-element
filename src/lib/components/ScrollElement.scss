@import url(https://fonts.googleapis.com/css?family=Josefin+Sans:300,400);

.scroll-element div {
  position: absolute;
  bottom: 20px;
  left: 50%;
  z-index: 2;
  display: inline-block;
  -webkit-transform: translate(0, -50%);
  transform: translate(0, -50%);
  color: #fff;
  font : normal 400 20px/1 'Josefin Sans', sans-serif;
  letter-spacing: .1em;
  text-decoration: none;
  transition: opacity .3s;
}
.scroll-element div:hover {
  opacity: .5;
}

.scroll-element-chevron {
  padding-top: 60px;
}
.scroll-element-chevron span {
  position: absolute;
  bottom: 5vh;
  left: 50%;
  width: 24px;
  height: 24px;
  margin-left: -12px;
  border-left: 1px solid #fff;
  border-bottom: 1px solid #fff;
  -webkit-transform: rotate(-45deg);
  transform: rotate(-45deg);
  box-sizing: border-box;
}

.scroll-element-chevron-sliding {
  padding-top: 60px;
}
.scroll-element-chevron-sliding span {
  position: absolute;
  top: 0;
  left: 50%;
  width: 24px;
  height: 24px;
  margin-left: -12px;
  border-left: 1px solid #fff;
  border-bottom: 1px solid #fff;
  -webkit-transform: rotate(-45deg);
  transform: rotate(-45deg);
  -webkit-animation: scroll-element-chevron-sliding 2s infinite;
  animation: scroll-element-chevron-sliding 2s infinite;
  box-sizing: border-box;
}
@-webkit-keyframes scroll-element-chevron-sliding {
  0% {
    -webkit-transform: rotate(-45deg) translate(0, 0);
  }
  20% {
    -webkit-transform: rotate(-45deg) translate(-10px, 10px);
  }
  40% {
    -webkit-transform: rotate(-45deg) translate(0, 0);
  }
}
@keyframes scroll-element-chevron-sliding {
  0% {
    transform: rotate(-45deg) translate(0, 0);
  }
  20% {
    transform: rotate(-45deg) translate(-10px, 10px);
  }
  40% {
    transform: rotate(-45deg) translate(0, 0);
  }
}

.scroll-element-chevron-sliding-fade {
  padding-top: 70px;
}
.scroll-element-chevron-sliding-fade span {
  position: absolute;
  top: 0;
  left: 50%;
  width: 24px;
  height: 24px;
  margin-left: -12px;
  border-left: 1px solid #fff;
  border-bottom: 1px solid #fff;
  -webkit-transform: rotate(-45deg);
  transform: rotate(-45deg);
  -webkit-animation: scroll-element-chevron-sliding-fade 1.5s infinite;
  animation: scroll-element-chevron-sliding-fade 1.5s infinite;
  box-sizing: border-box;
}
@-webkit-keyframes scroll-element-chevron-sliding-fade {
  0% {
    -webkit-transform: rotate(-45deg) translate(0, 0);
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    -webkit-transform: rotate(-45deg) translate(-20px, 20px);
    opacity: 0;
  }
}
@keyframes scroll-element-chevron-sliding-fade {
  0% {
    transform: rotate(-45deg) translate(0, 0);
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    transform: rotate(-45deg) translate(-20px, 20px);
    opacity: 0;
  }
}

.scroll-element-chevron-sliding-fade-triple {
  padding-top: 80px;
}
.scroll-element-chevron-sliding-fade-triple span {
  position: absolute;
  top: 0;
  left: 50%;
  width: 24px;
  height: 24px;
  margin-left: -12px;
  border-left: 1px solid #fff;
  border-bottom: 1px solid #fff;
  -webkit-transform: rotate(-45deg);
  transform: rotate(-45deg);
  -webkit-animation: scroll-element-chevron-sliding-fade-triple 2s infinite;
  animation: scroll-element-chevron-sliding-fade-triple 2s infinite;
  opacity: 0;
  box-sizing: border-box;
}
.scroll-element-chevron-sliding-fade-triple span:nth-of-type(1) {
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}
.scroll-element-chevron-sliding-fade-triple span:nth-of-type(2) {
  top: 16px;
  -webkit-animation-delay: .15s;
  animation-delay: .15s;
}
.scroll-element-chevron-sliding-fade-triple span:nth-of-type(3) {
  top: 32px;
  -webkit-animation-delay: .3s;
  animation-delay: .3s;
}
@-webkit-keyframes scroll-element-chevron-sliding-fade-triple {
  0% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
@keyframes scroll-element-chevron-sliding-fade-triple {
  0% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}

.scroll-element-chevron-sliding-fade-spin {
  padding-top: 70px;
}
.scroll-element-chevron-sliding-fade-spin span {
  position: absolute;
  top: 0;
  left: 50%;
  width: 24px;
  height: 24px;
  margin-left: -12px;
  border-left: 1px solid #fff;
  border-bottom: 1px solid #fff;
  -webkit-transform: rotateZ(-45deg);
  transform: rotateZ(-45deg);
  -webkit-animation: scroll-element-chevron-sliding-fade-spin 1.5s infinite;
  animation: scroll-element-chevron-sliding-fade-spin 1.5s infinite;
  box-sizing: border-box;
}
@-webkit-keyframes scroll-element-chevron-sliding-fade-spin {
  0% {
    -webkit-transform: rotateY(0) rotateZ(-45deg) translate(0, 0);
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    -webkit-transform: rotateY(720deg) rotateZ(-45deg) translate(-20px, 20px);
    opacity: 0;
  }
}
@keyframes scroll-element-chevron-sliding-fade-spin {
  0% {
    transform: rotateY(0) rotateZ(-45deg) translate(0, 0);
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    transform: rotateY(720deg) rotateZ(-45deg) translate(-20px, 20px);
    opacity: 0;
  }
}

.scroll-element-chevron-circle {
  padding-top: 60px;
}
.scroll-element-chevron-circle span {
  position: absolute;
  top: 0;
  left: 50%;
  width: 46px;
  height: 46px;
  margin-left: -23px;
  border: 1px solid #fff;
  border-radius: 100%;
  box-sizing: border-box;
}
.scroll-element-chevron-circle span::after {
  position: absolute;
  top: 50%;
  left: 50%;
  content: '';
  width: 16px;
  height: 16px;
  margin: -12px 0 0 -8px;
  border-left: 1px solid #fff;
  border-bottom: 1px solid #fff;
  -webkit-transform: rotate(-45deg);
  transform: rotate(-45deg);
  box-sizing: border-box;
}

.scroll-element-chevron-circle-pulse {
  padding-top: 60px;
}
.scroll-element-chevron-circle-pulse span {
  position: absolute;
  top: 0;
  left: 50%;
  width: 46px;
  height: 46px;
  margin-left: -23px;
  border: 1px solid #fff;
  border-radius: 100%;
  box-sizing: border-box;
}
.scroll-element-chevron-circle-pulse span::after {
  position: absolute;
  top: 50%;
  left: 50%;
  content: '';
  width: 16px;
  height: 16px;
  margin: -12px 0 0 -8px;
  border-left: 1px solid #fff;
  border-bottom: 1px solid #fff;
  -webkit-transform: rotate(-45deg);
  transform: rotate(-45deg);
  box-sizing: border-box;
}
.scroll-element-chevron-circle-pulse span::before {
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
  content: '';
  width: 44px;
  height: 44px;
  box-shadow: 0 0 0 0 rgba(255,255,255,.1);
  border-radius: 100%;
  opacity: 0;
  -webkit-animation: scroll-element-chevron-circle-pulse 3s infinite;
  animation: scroll-element-chevron-circle-pulse 3s infinite;
  box-sizing: border-box;
}
@-webkit-keyframes scroll-element-chevron-circle-pulse {
  0% {
    opacity: 0;
  }
  30% {
    opacity: 1;
  }
  60% {
    box-shadow: 0 0 0 60px rgba(255,255,255,.1);
    opacity: 0;
  }
  100% {
    opacity: 0;
  }
}
@keyframes scroll-element-chevron-circle-pulse {
  0% {
    opacity: 0;
  }
  30% {
    opacity: 1;
  }
  60% {
    box-shadow: 0 0 0 60px rgba(255,255,255,.1);
    opacity: 0;
  }
  100% {
    opacity: 0;
  }
}