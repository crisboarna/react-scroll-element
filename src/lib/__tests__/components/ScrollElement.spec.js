import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ScrollElement from '../../components/ScrollElement';

describe('ScrollElement', () => {
  let component;

  beforeAll(() => {
    configure({ adapter: new Adapter() });
  });

  describe('basic rendering', () => {
    it('should render  with skills array', () => {
      component = shallow(<ScrollElement />);
      expect(component.exists()).toEqual(true);
    });
  });

  describe('snapshots',() => {
    it('default component', () => {
      component = shallow(<ScrollElement/>);
      expect(component).toMatchSnapshot();
    });

    it('text', () => {
      component = shallow(<ScrollElement text='TEST'/>);
      expect(component).toMatchSnapshot();
    });

    describe('types',() => {
      it('chevron', () => {
        component = shallow(<ScrollElement type='chevron'/>);
        expect(component).toMatchSnapshot();
      });

      it('chevron sliding', () => {
        component = shallow(<ScrollElement type = 'chevron-sliding'/>);
        expect(component).toMatchSnapshot();
      });

      it('chevron sliding fade', () => {
        component = shallow(<ScrollElement type = 'chevron-sliding-fade'/>);
        expect(component).toMatchSnapshot();
      });

      it('chevron sliding fade triple', () => {
        component = shallow(<ScrollElement type = 'chevron-sliding-fade-triple'/>);
        expect(component).toMatchSnapshot();
      });

      it('chevron sliding fade spin', () => {
        component = shallow(<ScrollElement type = 'chevron-sliding-fade-spin'/>);
        expect(component).toMatchSnapshot();
      });

      it('chevron circle', () => {
        component = shallow(<ScrollElement type = 'chevron-circle'/>);
        expect(component).toMatchSnapshot();
      });

      it('chevron circle pulse', () => {
        component = shallow(<ScrollElement type = 'chevron-circle-pulse'/>);
        expect(component).toMatchSnapshot();
      });

      it('slider', () => {
        component = shallow(<ScrollElement type = 'slider'/>);
        expect(component).toMatchSnapshot();
      });

      it('slider chevron', () => {
        component = shallow(<ScrollElement type = 'slider-chevron'/>);
        expect(component).toMatchSnapshot();
      });

      it('slider action', () => {
        component = shallow(<ScrollElement type = 'slider-action'/>);
        expect(component).toMatchSnapshot();
      });
    });
  });
});