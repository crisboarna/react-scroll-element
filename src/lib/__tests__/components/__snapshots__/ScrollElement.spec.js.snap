// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollElement snapshots default component 1`] = `
<div
  className="scroll-element"
>
  <div
    className="scroll-element-chevron"
  >
    <span />
    Scroll
  </div>
</div>
`;

exports[`ScrollElement snapshots text 1`] = `
<div
  className="scroll-element"
>
  <div
    className="scroll-element-chevron"
  >
    <span />
    TEST
  </div>
</div>
`;

exports[`ScrollElement snapshots types chevron 1`] = `
<div
  className="scroll-element"
>
  <div
    className="scroll-element-chevron"
  >
    <span />
    Scroll
  </div>
</div>
`;

exports[`ScrollElement snapshots types chevron circle 1`] = `
<div
  className="scroll-element"
>
  <div
    className="scroll-element-chevron-circle"
  >
    <span />
    Scroll
  </div>
</div>
`;

exports[`ScrollElement snapshots types chevron circle pulse 1`] = `
<div
  className="scroll-element"
>
  <div
    className="scroll-element-chevron-circle-pulse"
  >
    <span />
    Scroll
  </div>
</div>
`;

exports[`ScrollElement snapshots types chevron sliding 1`] = `
<div
  className="scroll-element"
>
  <div
    className="scroll-element-chevron-sliding"
  >
    <span />
    Scroll
  </div>
</div>
`;

exports[`ScrollElement snapshots types chevron sliding fade 1`] = `
<div
  className="scroll-element"
>
  <div
    className="scroll-element-chevron-sliding-fade"
  >
    <span />
    Scroll
  </div>
</div>
`;

exports[`ScrollElement snapshots types chevron sliding fade triple 1`] = `
<div
  className="scroll-element"
>
  <div
    className="scroll-element-chevron-sliding-fade-triple"
  >
    <span />
    <span />
    <span />
    Scroll
  </div>
</div>
`;
